src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -25,8 +27,21 @@ public void printProducts() {
         }
     }
 
-    public void addProduct(Product shirt) {
-        products.add(shirt);
+    public void getAllProductsByCategory(Category category) {
+        for (Product product: products) {
+            if (product.getCategory() == category) {
+                System.out.println(product.getName());
+            }
+        }
+    }
+
+    public void addProduct(Product shirt) throws DuplicateProductException {
```

Comment:
i dont think this method will work correctly if you dont implement equals in Product